### PR TITLE
Add kernel testcases

### PR DESCRIPTION
**Pull request by NayanaNagaraj23**
The kernel testcases are modified to segregate it a positive and negative.

### CODE REVIEW

File: apps/examples/testcase/le_tc/kernel/tc_environ.c [hunk 1]
```diff
@@ -228,10 +260,14 @@ static void tc_environ_get_environ_ptr(void)
...
 int environ_main(void)
 {
-	tc_environ_setenv_getenv_unsetenv();
-	tc_environ_clearenv();
-	tc_environ_putenv();
-	tc_environ_get_environ_ptr();
+	tc_environ_setenv_getenv_unsetenv_pos();
+    tc_environ_getenv_neg();
+	tc_environ_setenv_name_null_neg();
+    tc_environ_setenv_empty_name_neg();
```
Comment (neel-samsung) [inline apps/examples/testcase/le_tc/kernel/tc_environ.c:266]: Please check indentation.

File: apps/examples/testcase/le_tc/kernel/itc_environ.c [hunk 2]
```diff
@@ -45,7 +45,7 @@
 * Preconditions         :none
 * Postconditions        :none
 */
-static void itc_environ_setenv_p(void)
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/itc_environ.c:48]: We check `_p` and `_n` internally in the automation tool. Do you have any reason to change?

File: apps/examples/testcase/le_tc/kernel/itc_libc_semaphore.c [hunk 3]
```diff
@@ -62,6 +62,6 @@ static void itc_libc_semaphore_getprotocol_n_no_sem_init(void)
 
 int itc_libc_semaphore_main(void)
 {
-	itc_libc_semaphore_getprotocol_n_no_sem_init();
+	itc_libc_semaphore_getprotocol_no_sem_init_neg();
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/itc_libc_semaphore.c:65]: same as previous
```
u/itc_module_feature_function_n/p_details
```
This is what we use in the naming and automation tool parses.
Please let me know why you change the naming.

File: apps/examples/testcase/le_tc/kernel/tc_clock.c [hunk 4]
```diff
@@ -184,10 +229,14 @@ static void tc_clock_clock_abstime2ticks(void)
...
-	tc_clock_clock_getres();
-	tc_clock_clock_abstime2ticks();
-
+	tc_clock_clock_gettimeofday_pos();
+	tc_clock_clock_gettimeofday_neg();
+	tc_clock_clock_get_time_invalid_clock_neg();
+	tc_clock_clock_get_time_pos();
+	tc_clock_clock_set_time_pos();
+	tc_clock_clock_set_time_neg();
+    tc_clock_clock_getres_pos();
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/tc_clock.c:238]: Please check indentation

File: apps/examples/testcase/le_tc/kernel/tc_libc_sched.c [hunk 5]
```diff
@@ -57,6 +54,18 @@ static void tc_libc_sched_sched_get_priority_max(void)
...
 }
 
+static void tc_libc_sched_sched_get_priority_max_sch_error_neg(void)
+{
+	int ret_chk;
+
+	ret_chk = sched_get_priority_max(SCHED_ERROR);
+	TC_ASSERT_EQ("sched_get_priority_max", ret_chk, ERROR);
+
+
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/tc_libc_sched.c:64]: Let's keep just 1 empty line for separating.

**Review by sunghan-chang — COMMENTED** [review]
I think you don't check the TizenRT commit rules. Could you check that at the wiki?

File: apps/examples/testcase/le_tc/kernel/itc_environ.c (hunk 2 above)
Comment (NayanaNagaraj23) [reply apps/examples/testcase/le_tc/kernel/itc_environ.c:48]: > We check `_p` and `_n` internally in the automation tool. Do you have any reason to change?

In some functions the _p and _n are at the middle and others are at the end of the function name, this is not identified properly in the TCM tool. So, I have changed the names as _pos and _neg at the end of the function for better identification and uniformity.

**Pull request closed without merging**